--- origsrc/llvm-2.9/Makefile.rules	2011-03-25 01:26:58.000000000 -0500
+++ src/llvm-2.9/Makefile.rules	2012-01-04 17:28:51.541900700 -0600
@@ -1030,6 +1030,9 @@ endif
 ifeq ($(HOST_OS), $(filter $(HOST_OS), Cygwin MingW))
 # LLVMLibsOptions is invalidated at processing tools/llvm-shlib.
 SharedLinkOptions += $(NativeExportsFile)
+ifndef LOADABLE_MODULE
+SharedLinkOptions += -Wl,--out-implib,$(SharedLibDir)/lib$(LIBRARYNAME).dll.a
+endif
 endif

 endif
@@ -1173,6 +1176,11 @@ install-local:: $(DestSharedLib)
 $(DestSharedLib): $(LibName.SO) $(DestSharedLibDir)
 	$(Echo) Installing $(BuildMode) Shared Library $(DestSharedLib)
 	$(Verb) $(INSTALL) $(LibName.SO) $(DestSharedLib)
+ifndef LOADABLE_MODULE
+ifeq ($(HOST_OS), $(filter $(HOST_OS), Cygwin MingW))
+	$(Verb) $(INSTALL) $(SharedLibDir)/lib$(LIBRARYNAME).dll.a $(DESTDIR)$(PROJ_libdir)
+endif
+endif

 uninstall-local::
 	$(Echo) Uninstalling $(BuildMode) Shared Library $(DestSharedLib)
