--- origsrc/llvm-3.3.src/test/Makefile	2013-05-04 07:36:23 +0000
+++ src/llvm-3.3.src/test/Makefile	2013-06-20 15:21:07 +0000
@@ -78,11 +78,15 @@ else # !SunOS
 ifeq ($(HOST_OS),AuroraUX)
 ULIMIT=ulimit -t 600 ; ulimit -d 512000 ; ulimit -v 512000 ;
 else # !AuroraUX
+ifeq (,$(findstring MINGW,$(uname_S)))
+ULIMIT=
+else # !MinGW
 # Newer versions of python try to allocate an insane amount of address space for
 # its thread-local storage, don't set a limit here.
 # When -v is not used, then -s has to be used to limit the stack size.
 # FIXME: Those limits should be enforced by lit instead of globally.
 ULIMIT=ulimit -t 600 ; ulimit -d 512000 ; ulimit -m 512000 ; ulimit -s 8192 ;
+endif # MinGW 
 endif # AuroraUX
 endif # SunOS
 
